subsys/bluetooth/shell/bt.c

```diff
@@ -1147,14 +1160,83 @@ static int cmd_scan_filter_clear_addr(const struct shell *sh, size_t argc,
 #endif /* CONFIG_BT_OBSERVER */
 
 #if defined(CONFIG_BT_BROADCASTER)
-static const struct bt_data ad_discov[] = {
-	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
-};
+static void ad_mode_parse(const struct shell *sh, size_t argc, char *argv[], bool *discov)
+{
+	for (size_t argn = 0; argn < argc; argn++) {
+		const char *arg = argv[argn];
+
+		if (!strcmp(arg, "discov")) {
+			*discov = true;
+			break;
+		} else if (!strcmp(arg, "non_discov")) {
+			*discov = false;
+			break;
+		}
+	}
+}
+
+static ssize_t ad_init(struct bt_data *data, const size_t data_size, const atomic_t *adv_opt)
```

Comment:
Does `data_size` here refer to the number of `bt_data` in the `data` array? Perhaps `size` should be replaced with `count`, `cnt` or `len`.

Reply:
It refers to the size of `data`, the number of entries that can be added. That about `data` -> `data_array` and `data_size` -> `data_array_size` ?

Reply:
That would work :) 

The term `size` is still a bit unclear, as `size` here, as you've said, refers to the number of entries in `data`, but it could just as easily refer to the actual size (in bytes) of the array. That's why I suggested count or length.